packages/cardano-services/test/Http/HttpServer.test.ts

```diff
@@ -531,4 +532,78 @@ describe('HttpServer', () => {
       expect(res.data).toMatchObject(meta);
     });
   });
+
+  describe('CORS', () => {
+    afterEach(async () => {
+      await httpServer.shutdown();
+    });
+
+    it('allows requests from all origins when allowed origins config is not specified', async () => {
+      httpServer = new HttpServer(
+        { allowedOrigins: undefined, listen: { host: 'localhost', port } },
+        {
+          logger,
+          runnableDependencies: [cardanoNode],
+          services: [new SomeHttpService(ServiceNames.Utxo, provider, logger)]
+        }
+      );
+
+      await httpServer.initialize();
+      await httpServer.start();
+      await onHttpServer(apiUrlBase);
+
+      const res = await axios.get(`${apiUrlBase}/health`, {
+        headers: { [CONTENT_TYPE]: 'application/json' }
+      });
+
+      expect(res.status).toBe(200);
+      expect(res.data).toBeDefined();
+    });
+
+    it('allows requests from whitelisted origins', async () => {
+      const allowedOrigin = 'http://cardano.com';
+
+      httpServer = new HttpServer(
+        { allowedOrigins: [allowedOrigin], listen: { host: 'localhost', port } },
+        {
+          logger,
+          runnableDependencies: [cardanoNode],
+          services: [new SomeHttpService(ServiceNames.Utxo, provider, logger)]
+        }
+      );
+
+      await httpServer.initialize();
+      await httpServer.start();
+      await onHttpServer(apiUrlBase);
+
+      const res = await axios.get(`${apiUrlBase}/${ServiceNames.Utxo}/health`, {
+        headers: { [CONTENT_TYPE]: 'application/json', [ORIGIN]: allowedOrigin }
+      });
+
+      expect(res.headers['access-control-allow-origin']).toEqual(allowedOrigin);
+    });
+
+    it('rejects requests from non-whitelisted origins', async () => {
```

Comment:
It's still allowing the requests. We could make it respond with 4xx to reject the request. WDYT @rhyslbw ?
```suggestion
    it('does not set "access-control-allow-origin" header for requests from non-whitelisted origins', async () => {
```

Reply:
I think we should not reject requests/throw an error in this case. The clients will be restricted from accessing shared resources if the `access-control-allow-origin` response header is not set (mirrored by the requested origin) which is enough @mkazlauskas

Reply:
> The clients will be restricted from accessing shared resources if the

I'm not sure it's a bad thing

Reply:
> I'm not sure it's a bad thing

Didn't get that one

Reply:
Implemented in d51f27c55b83c23829e4d00b33e5de92a9c660b2 @mkazlauskas